l2geth/rollup/sync_service.go

```diff
@@ -448,7 +469,36 @@ func (s *SyncService) updateL1GasPrice() error {
 		return err
 	}
 	s.RollupGpo.SetDataPrice(l1GasPrice)
-	log.Info("Adjusted L1 Gas Price", "gasprice", l1GasPrice)
+	return nil
+}
+
+// updateL2GasPrice will read from the L2 state the current L2 gas price.
+// It must be enabled to function until all nodes are running with the correct
+// contract deployed.
+func (s *SyncService) updateL2GasPrice(hash *common.Hash) error {
+	// TODO(mark): this is temporary and will be able to be rmoved when the
+	// OVM_GasPriceOracle is moved into the predeploy contracts
+	if !s.enableL2GasPolling {
+		return nil
+	}
```

Comment:
Go through entire codebase and convert execution price -> L2 gas price and data price to L1 gas price